doc/examples/custom_type.rst

```diff
@@ -0,0 +1,146 @@
+Custom Type Example
+===================
+
+This is an example of using a custom type with PyMongo. The example here shows
+how to subclass :class:`~bson.codec_options.TypeCodecBase` to write a type
+codec, which is used to populate a :class:`~bson.codec_options.TypeRegistry`.
+The type registry can then be used to create a custom-type-aware
+:class:`~pymongo.collection.Collection`. Read and write operations
+issued against the resulting collection object transparently manipulate
+documents as they are saved or retrieved from MongoDB.
+
+
+Setup
+-----
+
+We’ll start by getting a clean database to use for the example:
+
+.. doctest::
+
+  >>> from pymongo import MongoClient
+  >>> client = MongoClient()
+  >>> client.drop_database('custom_type_example')
+  >>> db = client.custom_type_example
+
+
+Since the purpose of the example is to demonstrate working with custom types,
+we’ll need a custom data type to use. For this example, we will be working with
+the :py:class:`~decimal.Decimal` type from Python's standard library. Since the
+BSON library has a :class:`~bson.decimal128.Decimal128` type (that implements
+the IEEE 754 decimal128 decimal-based floating-point numbering format) which
+is distinct from Python's built-in :py:class:`~decimal.Decimal` type, when we
+try to save an instance of ``Decimal`` with PyMongo, we get an
+:exc:`~bson.errors.InvalidDocument` exception.
+
+.. doctest::
+
+  >>> from decimal import Decimal
+  >>> mynumber = Decimal("45.321")
+  >>> db.test.insert_one({'mynumber': mynumber})
+  Traceback (most recent call last):
+  ...
+  bson.errors.InvalidDocument: Cannot encode object: <__main__.Decimal object at ...>
+
+
+.. _custom-type-type-codec:
+
+The Type Codec
+--------------
+
+In order to encode custom types, we must first define a **type codec** for our
+type. A type codec describes how an instance of a custom type can be
+*transformed* to/from one of the types :mod:`~bson` already understands, and
+can encode/decode. Type codecs must inherit from
+:class:`~bson.codec_options.TypeCodecBase`. Codecs that facilitate encoding of
+a custom type, must implement the ``python_type`` property, and the
```

Comment:
`Codecs that facilitate encoding of
a custom type, must implement ...` -> `In order to encode a custom type, a codec must implement the ``python_type`` property and the ``transform_python`` method`.

And same for the next sentence.